boards: nucleo_wba: Add flash partitions

erwango opened the pull request:
Add flash partitions which can be used with mcuboot.

erwango committed:
boards: nucleo_wba: Add flash partitions

Add flash partitions which can be used with mcuboot.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

gautierg-st commented on boards/arm/nucleo_wba52cg/nucleo_wba52cg.dts:
```diff
@@ -138,3 +139,32 @@ stm32_lp_tick_source: &lptim1 {
+		};
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(416)>;
```
Is there a reason why 416 and not 448 like slot1? Both spaces seem to be the same size (0x70000)

erwango replied:
No reason. Good catch, will fix.

gautierg-st commented on boards/arm/nucleo_wba55cg/nucleo_wba55cg.dts:
```diff
@@ -141,11 +142,25 @@ stm32_lp_tick_source: &lptim1 {
+		};
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_K(416)>;
```
Same question

nordicjm reviewed (dismissed):
Needs converting to hwmv2

erwango commented on boards/st/nucleo_wba55cg/nucleo_wba55cg.dts:
```diff
@@ -141,11 +142,25 @@ stm32_lp_tick_source: &lptim1 {
+			label = "image-1";
+			reg = <0x00080000 DT_SIZE_K(448)>;
+		};
+		scratch_partition: partition@f0000 {
+			label = "image-scratch";
+			reg = <0x000f0000 DT_SIZE_K(16)>;
+		};
```
ok, thanks. I think we should do this widely on all STM32 targets at once. 
From what I see `MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH` is actually the default, so these are not in use anyway. I'll trigger the change.

erwango committed:
boards: st: nucleo_wba: Fix openocd configuration

Using Zephyr SDK 0.16.5, openocd can be used to debug WBA based boards.
Fix documentation and clean up board.cmake configuration for
nucleo_wba52cg from a local configuration which shouldn't have been
upstreamed

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>
